pkg/kubectl/resource_filter.go

```diff
@@ -57,7 +57,7 @@ func filterPods(obj runtime.Object, options PrintOptions) bool {
 }
 
 // Filter loops through a collection of FilterFuncs until it finds one that can filter the given resource
-func (f Filters) Filter(obj runtime.Object, opts *PrintOptions) (bool, error) {
+func (f Filters) Filter(obj runtime.Object, opts *PrintOptions) (bool, runtime.Object, error) {
 	// check if the object is unstructured. If so, let's attempt to convert it to a type we can understand
 	// before apply filter func.
 	switch obj.(type) {
```

Comment:
Thanks for your suggestion, it's really better to abstract the decode part into a function. I've done the modifications and fix the test failures. Please check for this.